### ISSUE TITLE
HTTP.request with body::Vector{Any} throws an exception 

### ISSUE THREAD

**Issue by eduartos** [JuliaWeb/HTTP.jl#895]
**Versions**
 - Julia 1.7.3
 - HTTP.jl 1.2.0

Sending a request with `body = Vector{Any}()` throws an exception:

```
Jul 26 13:29:02: ┌ Error: unhandled exception
Jul 26 13:29:02: │   exception =
Jul 26 13:29:02: │    MethodError: no method matching ismarked(::Vector{Any})
Jul 26 13:29:02: │    Closest candidates are:
Jul 26 13:29:02: │      ismarked(!Matched::Base.AbstractPipe) at io.jl:380
Jul 26 13:29:02: │      ismarked(!Matched::Base.LibuvStream) at stream.jl:1268
Jul 26 13:29:02: │      ismarked(!Matched::TranscodingStreams.TranscodingStream) at /root/.julia/packages/TranscodingStreams/IVlnc/src/stream.jl:202
Jul 26 13:29:02: │      ...
Jul 26 13:29:02: │   stack =
Jul 26 13:29:02: │    50-element Vector{Base.StackTraces.StackFrame}:
Jul 26 13:29:02: │     retryable(r::HTTP.Messages.Request) at Messages.jl:264
Jul 26 13:29:02: │     (::HTTP.RetryRequest.var"#3#6"{Int64, HTTP.Messages.Request, Base.RefValue{Int64}})(s::Tuple{Int64, Float64}, ex::HTTP.Exceptions.RequestError) at RetryRequest.jl:42
Jul 26 13:29:02: │     ⋮
```

**Comment by stelmo** [comment]
I also get a similar error. Here is a MWE (I need to run it twice to get the error):
```julia
body = 
  """
      PREFIX rh: <http://rdf.rhea-db.org/>
      PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
      SELECT ?verb ?obj 
      WHERE {
        rh:11844 ?verb ?obj .
      }
  """
r = HTTP.request(
    "POST",
    endpoint_url,
    [
        "Accept" => "application/sparql-results+json",
        "Content-type" => "application/x-www-form-urlencoded",
    ],
    Dict("query" => body),
)
```
with error:
```julia
julia> r = HTTP.request(                                                                                                                                                       
           "POST",                                                                                                                                                             
           endpoint_url,                                                                                                                                                       
           [                                                                                                                                                                   
               "Accept" => "application/sparql-results+json",                                                                                                                  
               "Content-type" => "application/x-www-form-urlencoded",                                                                                                          
                   ],                                                                                                                                                          
           Dict("query" => body),                                                                                                                                              
           )
ERROR: MethodError: no method matching ismarked(::Dict{String, String})
Closest candidates are:
  ismarked(::Base.AbstractPipe) at io.jl:427
  ismarked(::Base.LibuvStream) at stream.jl:1458
  ismarked(::TranscodingStreams.TranscodingStream) at C:\Users\stelmo\.julia\packages\TranscodingStreams\IVlnc\src\stream.jl:202
  ...
```
I am in Julia v1.8.0-rc3 and HTTP v1.2.0.

**Comment by quinnj** [comment]
Fixed in #904

**Comment by stelmo** [comment]
Thank you very much!!
